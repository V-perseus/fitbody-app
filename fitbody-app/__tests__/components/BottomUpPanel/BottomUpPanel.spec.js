import React from 'react'
import { Text } from 'react-native'
import { render } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

import BottomPanelUp from '../../../src/components/BottomUpPanel'

jest.useFakeTimers()

describe('BottomPanelUp', () => {
  it('renders snapshot correctly', () => {
    const tree = renderer
      .create(
        <BottomPanelUp
          isOpen={true}
          startHeight={0}
          topEnd={400}
          header={() => null}
          content={() => <Text testID="content">content</Text>}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render content', async () => {
    const { queryByTestId, queryByText, getByTestId } = render(
      <BottomPanelUp
        isOpen={true}
        startHeight={0}
        topEnd={400}
        header={() => null}
        content={() => <Text testID="content">content</Text>}
      />,
    )
    await getByTestId('content')
    // await getByTestId('scrollview')
    // expect(getByTestId('scrollview')).not.toBeNull()
    expect(queryByText('content')).not.toBeNull()
  })

  // it('should handle toggle press event', async () => {
  //   const { container, getByTestId, getByText } = render(
  //     <BottomPanelUp isOpen={false} startHeight={0} topEnd={400} header={() => <Text>toggle</Text>} content={() => null} />,
  //   )
  //   expect(container.instance.state.open).toEqual(false)
  //   await fireEvent.press(getByText('toggle'))
  //   expect(container.instance.state.open).toEqual(true)
  // })
})
