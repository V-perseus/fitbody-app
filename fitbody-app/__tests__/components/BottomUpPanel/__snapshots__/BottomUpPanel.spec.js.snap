// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomPanelUp renders snapshot correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
      },
      {},
    ]
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "paddingBottom": 400,
        "position": "absolute",
        "right": 0,
        "transform": [
          {
            "translateY": 400,
          },
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={12}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "alignItems": "center",
            "height": -117,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 24,
              "elevation": 21,
              "height": 48,
              "justifyContent": "center",
              "marginBottom": -24,
              "width": 48,
              "zIndex": 10,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "borderColor": "#bbbbbb",
                "borderRadius": 17.5,
                "borderWidth": 1.7,
                "height": 35,
                "justifyContent": "center",
                "width": 35,
              }
            }
          >
            <close.svg
              color="#bbbbbb"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={{}}
    >
      <RCTScrollView
        bounces={false}
        contentContainerStyle={
          {
            "backgroundColor": "#ffffff",
            "flexGrow": 1,
          }
        }
        indicatorStyle="black"
        scrollEnabled={true}
        scrollEventThrottle={16}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
      >
        <View>
          <Text
            testID="content"
          >
            content
          </Text>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
